tests/functional/c/consider/consider_using_str_partition.py

```diff
@@ -0,0 +1,81 @@
+"""Emit a message for iteration through dict keys and subscripting dict with key."""
+# pylint: disable=line-too-long,missing-docstring,unsubscriptable-object,too-few-public-methods,invalid-name,redefined-builtin
+
+# Test subscripting .split()
+get_first = '1,2,3'.split(',')[0]  # [consider-using-str-partition]
+get_last = '1,2,3'[::-1].split(',')[0]  # [consider-using-str-partition]
+
+SEQ = '1,2,3'
+get_first = SEQ.split(',')[0]  # [consider-using-str-partition]
+get_last = SEQ.split(',')[-1]  # [consider-using-str-partition]
+get_first = SEQ.rsplit(',')[0]  # [consider-using-str-partition]
+get_last = SEQ.rsplit(',')[-1]  # [consider-using-str-partition]
+
+get_mid = SEQ.split(',')[1]  # This is okay
+get_mid = SEQ.split(',')[-2]  # This is okay
+
+
+# Test varying maxsplit argument
+## str.split() tests
+bad_split = '1,2,3'.split(sep=',', maxsplit=1)[1]  # [consider-using-str-partition]
+
+good_split = '1,2,3'.split(sep=',', maxsplit=1)[-1]  # This is fine
+good_split = '1,2,3'.split(sep=',', maxsplit=1)[0]  # This is fine

```

Comment:
Why doesn't it make sense to suggest `partition(',')[0]` here?

Reply:
so in this case, we should suggest partition/rpartition as long as ``maxsplit == 1``?

I thought it would be okay as it's computationally equivalent to partition.

Reply:
I will resolve this then :)